common/thrift_router.h

```diff
@@ -367,6 +371,7 @@ class ThriftRouter {
      * hosts first by
      * (1) Prefer master to slave, then (2) Prefer local to non-local.
      *
+     * Otherwise, if az is specified by caller, return hosts belonging to that az
```

Comment:
comments here is a bit confusing, Can you update them to be more clear, now that we have three conditions 
- AZ aware
- prefer master, then prefer local
- prefer local only

Reply:
added the comments now